curl: 8.4.0 -> 8.5.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/project/curl/versions.


meta.description for curl is: A command line tool for transferring files with URL syntax

meta.homepage for curl is: https://curl.se/

meta.changelog for curl is: https://curl.se/changes.html#8_5_0


###### Updates performed
- Version update

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- Warning: a test defined in `passthru.tests` did not pass
- found 8.5.0 with grep in /nix/store/41xjs6ssq20w37na9mq23jyvyazmafd6-curl-8.5.0-bin
- found 8.5.0 in filename of file in /nix/store/41xjs6ssq20w37na9mq23jyvyazmafd6-curl-8.5.0-bin

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
48118 total rebuild path(s)

48118 package rebuild(s)

First fifty rebuilds by attrpath

ArchiSteamFarm
BeatSaberModManager
CHOWTapeModel
ChowCentaur
ChowKick
ChowPhaser
CuboCore.coreaction
CuboCore.corearchiver
CuboCore.corefm
CuboCore.coregarage
CuboCore.corehunt
CuboCore.coreimage
CuboCore.coreinfo
CuboCore.corekeyboard
CuboCore.corepad
CuboCore.corepaint
CuboCore.corepdf
CuboCore.corepins
CuboCore.corerenamer
CuboCore.coreshot
CuboCore.corestats
CuboCore.corestuff
CuboCore.coreterminal
CuboCore.coretime
CuboCore.coretoppings
CuboCore.coreuniverse
CuboCore.libcprime
CuboCore.libcsys
DisnixWebService
EBTKS
EmptyEpsilon
Fabric
HentaiAtHome
LAStools
LASzip
LASzip2
LibreArp
LibreArp-lv2
Literate
MIDIVisualizer
MMA
NSPlist
OSCAR
OVMF
OVMFFull
PageEdit
PlistCpp
QuadProgpp
R
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/41xjs6ssq20w37na9mq23jyvyazmafd6-curl-8.5.0-bin \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A curl https://github.com/r-ryantm/nixpkgs/archive/7be9f0f52630eec0bd9026ff0e0e31c8116ac7b2.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/7be9f0f52630eec0bd9026ff0e0e31c8116ac7b2#curl
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/41xjs6ssq20w37na9mq23jyvyazmafd6-curl-8.5.0-bin
ls -la /nix/store/41xjs6ssq20w37na9mq23jyvyazmafd6-curl-8.5.0-bin/bin
```

---

</details>
<br/>



### Pre-merge build results

NixPkgs review skipped

---

###### Maintainer pings

cc @lovek323 for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

r-ryantm committed:
curl: 8.4.0 -> 8.5.0

panicgh commented:
There are 2 known vulnerabilities in 8.4.0: https://curl.se/docs/vuln-8.4.0.html

mweinelt commented:
Builds on aarch64-{darwin,linux}, x86_64-{darwin,linux}.

mweinelt committed:
python311Packages.pycurl: disable failing test

We already disabled another test due to this issue, so I'm confident
this is the correct way forward.

mweinelt commented:
@ofborg eval
